modules/ting/src/Search/DingProviderStrategy.php

```diff
@@ -25,7 +25,13 @@ class DingProviderStrategy implements TingSearchStrategyInterface {
    *   The result of performing the search.
    */
   public function executeSearch($query) {
-    return ding_provider_invoke('search', 'search', $query);
+    try {
+      return ding_provider_invoke('search', 'search', $query);
+    }
+    catch (\Exception $e) {
```

Comment:
Catching the generic `\Exception` class is very broad and thus dangerous. I would much prefer if we used a generic search/provider/whatever exception class which could be thrown/caught in these situations.